### PR TITLE
high performance job get

### PR DESCRIPTION
**Pull request by 8olio8**

### CODE REVIEW

File: pkg/apiserver/models/job.go [hunk 1]
```diff
@@ -87,6 +89,37 @@ func (job *Job) BeforeSave(tx *gorm.DB) error {
 	return nil
 }
 
+func (job *Job) decode() error {
```
Comment (D0m021ng) [inline pkg/apiserver/models/job.go]: use `AfterFind` or `BeforeSave`

File: pkg/apiserver/models/job_task.go [hunk 2]
```diff
@@ -74,3 +76,29 @@ func UpdateTask(task *JobTask) error {
 	}).Create(task)
 	return tx.Error
 }
+
+func (j *JobTask) decode() error {
```
Comment (D0m021ng) [inline pkg/apiserver/models/job_task.go]: Use `BeforeSave` and `AfterFind`

File: pkg/apiserver/controller/job/job.go [hunk 3]
```diff
@@ -32,6 +36,88 @@ import (
...
 
+var (
+	WSManager = WebsocketManager{
+		Connections:   make(map[string]*Connection),
+		BroadcastChan: make(chan GetJobResponse, 1000),
+	}
+	UpdateTime = time.Now()
+)
+
+type JobSpec struct {
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job.go]: Repeated with `CreateSingleJobRequest` at line 122.

File: pkg/apiserver/controller/job/job_conn.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+type MsgType string
+
+const (
+	HeartbeatMsg MsgType = "HeartbeatMsg"
+	DataMsg      MsgType = "DataMsg"
+)
+
+var (
+	timeLayoutStr = "2006-01-02 15:04:05"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job_conn.go]: Repeated with  `pkg/apiserver/models/commom.TimeFormat`

File: pkg/apiserver/controller/job/ws_manager.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
+package job
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/ws_manager.go]: Please use `pkg/commom/uuid`

File: pkg/apiserver/common/errors.go [hunk 6]
```diff
@@ -80,6 +80,8 @@ const (
 	RunCacheNotFound      = "RunCacheNotFound"
 	ArtifactEventNotFound = "ArtifactEventNotFound"
 
+	JobNotFound = "JobNotFound"
+
```
Comment (qiaoshuangshuang) [inline pkg/apiserver/common/errors.go:84]: go_fmt

**Pull request closed without merging**